django/db/migrations/executor.py

```diff
@@ -329,8 +329,11 @@ def should_skip_detecting_model(migration, model):
         apps = after_state.apps
         found_create_model_migration = False
         found_add_field_migration = False
+        ignores_table_name_case = self.connection.features.ignores_table_name_case
         with self.connection.cursor() as cursor:
-            existing_table_names = self.connection.introspection.table_names(cursor)
+            existing_table_names = set(self.connection.introspection.table_names(cursor))
+            if ignores_table_name_case:
+                existing_table_names = [name.casefold() for name in existing_table_names]
```

Comment:
You'll want to make this a set as well

```suggestion
                existing_table_names = {name.casefold() for name in existing_table_names}
```